packages/atomic/src/components/commerce/facets/atomic-commerce-timeframe-facet/atomic-commerce-timeframe-facet.tsx

```diff
@@ -0,0 +1,317 @@
+    const displayValue = this.formatFacetValue(facetValue);
+    const isSelected = facetValue.state === 'selected';
+    const isExcluded = facetValue.state === 'excluded';
+    return (
+      <FacetValueLink
+        displayValue={displayValue}
+        isSelected={isSelected}
+        numberOfResults={facetValue.numberOfResults}
+        i18n={this.bindings.i18n}
+        onClick={() => this.facet.toggleSingleSelect(facetValue)}
+      >
+        <FacetValueLabelHighlight
+          displayValue={displayValue}
+          isSelected={isSelected}
+          isExcluded={isExcluded}
+        ></FacetValueLabelHighlight>
+      </FacetValueLink>
+    );
+  }
+
+  private renderValuesContainer(children: VNode[]) {
+    return (
+      <FacetValuesGroup i18n={this.bindings.i18n} label={this.displayName}>
+        <ul class="mt-3" part="values">
+          {children}
+        </ul>
+      </FacetValuesGroup>
+    );
+  }
+
+  private renderHeader() {
+    return (
+      <FacetHeader
+        i18n={this.bindings.i18n}
+        label={this.displayName}
+        onClearFilters={() => {
+          this.focusTarget.focusAfterSearch();
+          if (this.hasInputRange) {
+            this.resetRange();
+            return;
+          }
+          this.facet.deselectAll();
+        }}
+        numberOfActiveValues={this.numberOfSelectedValues}
+        isCollapsed={this.isCollapsed}
+        headingLevel={0}
+        onToggleCollapse={() => (this.isCollapsed = !this.isCollapsed)}
+        headerRef={(el) => this.focusTarget.setTarget(el)}
+      ></FacetHeader>
+    );
+  }
+
+  private renderDateInput() {
+    return (
+      <atomic-facet-date-input
+        bindings={this.bindings}
+        label={this.displayName}
+        rangeGetter={() => this.inputRange}
+        facetId={this.facet.state.facetId}
+        rangeSetter={({start, end, endInclusive}: DateRangeRequest) => {
+          this.facet.setRanges([
+            {
+              start,
+              end,
+              endInclusive,
+              isAutoSelected: false,
+              isSuggested: false,
+              numberOfResults: 0,
+              moreValuesAvailable: false,
+              state: 'selected',
+            },
+          ]);
+        }}
+      ></atomic-facet-date-input>
+    );
+  }
+
+  public render() {
+    const {hasError, firstSearchExecuted} = this.summary.state;
+
+    return (
+      <FacetGuard
+        enabled={true}
+        firstSearchExecuted={firstSearchExecuted}
+        hasError={hasError}
+        hasResults={this.shouldRenderFacet}
+      >
+        {firstSearchExecuted ? (
+          <FacetContainer>
+            {this.renderHeader()}
+            {!this.isCollapsed && [
+              this.shouldRenderValues && this.renderValues(),
+              this.shouldRenderInput && this.renderDateInput(),
+            ]}
+          </FacetContainer>
+        ) : (
+          <FacetPlaceholder isCollapsed={this.isCollapsed} numberOfValues={8} />
```

Comment:
Should we set numberOfValues to `facetState.values.length` when it's greater than 0, and fallback to 8 (or 5?) otherwise?